src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -57,22 +57,27 @@ static ssize_t hwmon_sensor_show(struct device *dev,
 		return sprintf(buf, "%d\n", 0);
 	}
 
-	switch(field_id) {
-	case SENSOR_NAME:
+	if (field_id == SENSOR_NAME) {
 		memcpy(output, &sdm->sensor_data[repo_id][buf_index], buf_len);
 		return snprintf(buf, buf_len + 2, "%s\n", output);
-	case SENSOR_VALUE:
+	}
+
+	if (field_id == SENSOR_VALUE) {
 		memcpy(value, &sdm->sensor_data[repo_id][buf_index], buf_len);
 		return sprintf(buf, "%u\n", *value);
-	case SENSOR_AVG_VAL:
+	}
+
+	if (field_id == SENSOR_AVG_VAL) {
 		memcpy(value, &sdm->sensor_data[repo_id][buf_index], buf_len);
 		return sprintf(buf, "%u\n", *value);
-	case SENSOR_MAX_VAL:
+	}
+
+	if (field_id == SENSOR_MAX_VAL) {
 		memcpy(value, &sdm->sensor_data[repo_id][buf_index], buf_len);
 		return sprintf(buf, "%u\n", *value);
-	default:
-		return sprintf(buf, "%d\n", 0);
 	}
+
+	return sprintf(buf, "N/A\n");
```

Comment:
why do you need to change switch statement to if-else statement here?

Reply:
Hi @maxzhen switch statement expecting "break", but we used return. So, hence the issue.